frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -485,42 +474,37 @@ export default ({
           contractID,
           data,
           hooks: {
-            preSendCheck: (message, state) => {
+            preSendCheck: async (message, state) => {
               // NOTE: this preSendCheck does nothing except appending a pending message
               //       temporarily until the uploading attachments is finished
               //       it always returns false, so it doesn't affect the contract state
-              const [, opV] = message.op()
-              const { meta } = opV.valueOf().valueOf()
-
-              temporaryMessage = createMessage({
-                meta,
-                data: { ...data, attachments },
-                hash: message.hash(),
-                height: message.height(),
-                state: this.messageState.contract,
-                pending: true,
-                innerSigningContractID: this.ourIdentityContractId
-              })
-              this.messages.push(temporaryMessage)
-
               this.stopReplying()
               this.updateScroll()
+
+              Vue.set(this.messageState, 'contract', await sbp('chelonia/in/processMessage', message, this.messageState.contract))
+              temporaryMessage = this.messages.find((m) => m.hash === message.hash())
+
               return false
             }
           }
         }).then(async () => {
-          const isUploaded = await uploadAttachments()
-          if (isUploaded) {
-            const removeTemporaryMessage = () => {
-              // NOTE: remove temporary message which is created before uploading attachments
-              if (temporaryMessage) {
-                const msgIndex = findMessageIdx(temporaryMessage.hash, this.messages)
-                this.messages.splice(msgIndex, 1)
-              }
+          await uploadAttachments()
```

Comment:
Why are we no longer caring about the return value?

Reply:
Good question. Maybe the question can also be why did we care before? `uploadAttachments` always returns true.

https://github.com/okTurtles/group-income/blob/4464feab187308af2cfeefa6a73ed8f5161f99a3/frontend/views/containers/chatroom/ChatMain.vue#L473